src/components.js

```diff
@@ -0,0 +1,117 @@
+import store from './todo-store.js';
+
+function iconButton(name) {
+  const node = document.createElement('button');
+  node.classList.add('icon-button');
+
+  const icon = document.createElement('span');
+  icon.innerText = name;
+  icon.classList.add('material-icons', 'icon');
+
+  node.appendChild(icon);
+  return node;
+}
+
+export function listItemComponent({
+  index, description, completed, onToggle, onSwap, onEdit, onDelete,
+}) {
+  const node = document.createElement('li');
+  node.classList.add('todo');
+
+  // Drag and Drop listeners
+  node.addEventListener('dragstart', (event) => {
+    node.classList.add('dragging');
+    setTimeout(() => {
+      node.classList.add('opacity-0');
+    }, 0);
+    event.dataTransfer.setData('index', index);
```

Comment:
- [ ] It looks like you add an index on your elements, but I'm afraid I couldn't see it for each element on HTML code, please, add it. 🐶
![image](https://user-images.githubusercontent.com/20563252/130127085-7a6311dc-5e27-4542-a057-298fcad7a31d.png)

Reply:
You should add the index to each `todo`, so that you know which one to delete when removing one, at the moment, it deleted the last `todo` even though I'm picking other ones